packages/core/test/util/renderer.test.ts

```diff
@@ -588,3 +589,100 @@ test('mapStateToLayoutProps - hidden via state with path from ownProps ', t => {
   const props = mapStateToLayoutProps(state, ownProps);
   t.false(props.visible);
 });
+
+test('should assign defaults to enum', t => {
+  const schema: JsonSchema = {
+    type: 'object',
+    properties: {
+      name: {
+        type: 'string',
+        minLength: 1
+      },
+      color: {
+        type: 'string',
+        enum: ['red', 'green', 'blue'],
+        default: 'green'
+      }
+    }
+  };
+
+  const uischema: UISchemaElement = undefined;
+
+  const data = {
+    name: 'foo'
+  };
+
+  const initState = {
+    jsonforms: {
+      core: {
+        uischema,
+        schema,
+        data,
+        errors: [] as ErrorObject[]
+      }
+    }
+  };
+  const store: Store<JsonFormsState> = createStore(
+    combineReducers({ jsonforms: jsonformsReducer() }),
+    initState
+  );
+  store.dispatch(
+    init(data, schema, uischema, createAjv({ useDefaults: true }))
+  );
+  const ownProps: OwnPropsOfControl = { schema };
+
+  mapDispatchToArrayControlProps(store.dispatch, ownProps);
+
+  t.is(store.getState().jsonforms.core.data.color, 'green');
+});
+
+test('should assign defaults to array', t => {
+  const schema: JsonSchema = {
+    type: 'array',
+    items: {
+      type: 'object',
+      properties: {
+        message: {
+          type: 'string',
+          default: 'foo'
+        }
+      }
+    }
+  };
+
+  const uischema: ControlElement = {
+    type: 'Control',
+    scope: '#'
+  };
+
+  const data = [{}];
+
+  const initState = {
+    jsonforms: {
+      core: {
+        uischema,
+        schema,
+        data,
+        errors: [] as ErrorObject[]
+      }
+    }
+  };
+  const store: Store<JsonFormsState> = createStore(
+    combineReducers({ jsonforms: jsonformsReducer() }),
+    initState
+  );
+  store.dispatch(
+    init(data, schema, uischema, createAjv({ useDefaults: true }))
+  );
+  const ownProps: OwnPropsOfControl = {
+    schema,
+    uischema
+  };
+  const props = mapDispatchToArrayControlProps(store.dispatch, ownProps);
+
+  props.addItem('')();
+
+  t.is(store.getState().jsonforms.core.data.length, 2);
+  t.deepEqual(store.getState().jsonforms.core.data[0], { message: 'foo' });
```

Comment:
Split the expectation of having the default initialization kick-in for the 1st element into a separate test case. This test should only concern itself with checking the defaults of a newly added item.

Reply:
Added a new test.